From bb0b1ea6b7445ddcf6c681876d338d34eb917513 Mon Sep 17 00:00:00 2001
From: Christoph Jabs <christoph.jabs@helsinki.fi>
Date: Tue, 1 Oct 2024 12:36:03 +0300
Subject: [PATCH] perf: disable finalize

---
 src/internal.cpp | 91 ++++++++++++++++++++++++------------------------
 1 file changed, 46 insertions(+), 45 deletions(-)

diff --git a/src/internal.cpp b/src/internal.cpp
index d524edd..d450a54 100644
--- a/src/internal.cpp
+++ b/src/internal.cpp
@@ -841,52 +841,53 @@ void Internal::finalize (int res) {
   if (!proof)
     return;
   LOG ("finalizing");
-  // finalize external units
-  for (const auto &evar : external->vars) {
-    assert (evar > 0);
-    const auto eidx = 2 * evar;
-    int sign = 1;
-    uint64_t id = external->ext_units[eidx];
-    if (!id) {
-      sign = -1;
-      id = external->ext_units[eidx + 1];
-    }
-    if (id) {
-      proof->finalize_external_unit (id, evar * sign);
-    }
-  }
-  // finalize internal units
-  for (const auto &lit : lits) {
-    const auto elit = externalize (lit);
-    if (elit) {
-      const unsigned eidx = (elit < 0) + 2u * (unsigned) abs (elit);
-      const uint64_t id = external->ext_units[eidx];
-      if (id) {
-        assert (unit_clauses[vlit (lit)] == id);
-        continue;
-      }
-    }
-    const auto uidx = vlit (lit);
-    const uint64_t id = unit_clauses[uidx];
-    if (!id)
-      continue;
-    proof->finalize_unit (id, lit);
-  }
-  // See the discussion in 'propagate' on why garbage binary clauses stick
-  // around.
-  for (const auto &c : clauses)
-    if (!c->garbage || c->size == 2)
-      proof->finalize_clause (c);
-
-  // finalize conflict and proof
-  if (conflict_id) {
-    proof->finalize_clause (conflict_id, {});
-  }
+  // // finalize external units
+  // for (const auto &evar : external->vars) {
+  //   assert (evar > 0);
+  //   const auto eidx = 2 * evar;
+  //   int sign = 1;
+  //   uint64_t id = external->ext_units[eidx];
+  //   if (!id) {
+  //     sign = -1;
+  //     id = external->ext_units[eidx + 1];
+  //   }
+  //   if (id) {
+  //     proof->finalize_external_unit (id, evar * sign);
+  //   }
+  // }
+  // // finalize internal units
+  // for (const auto &lit : lits) {
+  //   const auto elit = externalize (lit);
+  //   if (elit) {
+  //     const unsigned eidx = (elit < 0) + 2u * (unsigned) abs (elit);
+  //     const uint64_t id = external->ext_units[eidx];
+  //     if (id) {
+  //       assert (unit_clauses[vlit (lit)] == id);
+  //       continue;
+  //     }
+  //   }
+  //   const auto uidx = vlit (lit);
+  //   const uint64_t id = unit_clauses[uidx];
+  //   if (!id)
+  //     continue;
+  //   proof->finalize_unit (id, lit);
+  // }
+  // // See the discussion in 'propagate' on why garbage binary clauses
+  // stick
+  // // around.
+  // for (const auto &c : clauses)
+  //   if (!c->garbage || c->size == 2)
+  //     proof->finalize_clause (c);
+
+  // // finalize conflict and proof
+  // if (conflict_id) {
+  //   proof->finalize_clause (conflict_id, {});
+  // }
   proof->report_status (res, conflict_id);
-  if (res == 10)
-    external->conclude_sat ();
-  else if (res == 20)
-    conclude_unsat ();
+  // if (res == 10)
+  //   external->conclude_sat ();
+  // else if (res == 20)
+  //   conclude_unsat ();
 }
 
 /*------------------------------------------------------------------------*/
-- 
2.46.0

